server/app/controllers/fusor/api/v21/subscriptions_controller.rb

```diff
@@ -13,6 +13,19 @@
 module Fusor
   class Api::V21::SubscriptionsController < Api::V2::BaseController
 
+    before_filter :check_content_type, :only => :upload
+
+    def check_content_type
+      # overwrite this method in V2::BaseController that
+      # renders error if request.content_type != "application/json"
+      # the upload method uses "multipart/form-data"
+    end
+
+    def upload
+      # placeholder only. TODO perform actually upload file
+      render :json => {params: params, request: @request}
```

Comment:
I have that already, except my render is different. I'll update my end to match. I just had render :json => {}